generate-styles.js

```diff
@@ -14,8 +31,7 @@ function linkToGithubUsername(rawLink) {
 		return null;
 	}
 
-	// truncate to avoid maliciously excessively long URLs
-	let link = url.parse(rawLink.slice(0, 100));
+	let link = url.parse(rawLink);
```

Comment:
Why was this changed? Just curious

Reply:
If one knows how much data we take in, they could write:

"gh.com/<target>/////{100 times}////////../<actual user>"

It would be resolved as <target> in our script but as <actual user> in the link.

Reply:
I don't think we need to be so concerned about failures to detect github usernames, false negatives. After all, if someone doesn't want this to identify them, they could simply make their link be one to a personal website, redirect server, social media profile, etc. 

I am more concerned about attacks on the script based on vulnerabilities. Such as the CSS injection vulnerability we saw earlier. I suspect that the slicing was originally introduced to avoid some sort of DOS attack?

Reply:
stick was already targeted with such an attack and I would prefer that it doesn't happen again.

The slicing made sense before I realized that github is awfully compliant on its URL spec implementation.

The only realistic DoS that i can envision happening rn would be a github action runner hanging, which doesn't seem like the most interesting target in the world. 
I do envision another tho, I'll think about how to prevent this one.

I would however want to reintroduce some checks on the length and the characters of the github usernames, which I forgot to get back in the code. This should be enough, in my opinion.